src/common/page_table.cpp

```diff
@@ -10,11 +10,65 @@ PageTable::PageTable() = default;
 
 PageTable::~PageTable() noexcept = default;
 
-void PageTable::Resize(size_t address_space_width_in_bits, size_t page_size_in_bits) {
-    const size_t num_page_table_entries{1ULL << (address_space_width_in_bits - page_size_in_bits)};
+bool PageTable::BeginTraversal(TraversalEntry* out_entry, TraversalContext* out_context,
```

Comment:
```suggestion
bool PageTable::BeginTraversal(TraversalEntry& out_entry, TraversalContext& out_context,
```

Since it's a logic bug otherwise to use a null pointer for either of these